### DISCUSSION TITLE
Before You Blame The App

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#511] [Show and tell]
Cruiser and I have a good relationship - although it's not without the occasional "hiccup".

Like, [when I couldn't get it to start a route on a roundabout](https://github.com/devemux86/cruiser/discussions/150).
Or when it [wouldn't route me over a certain type of road furniture](https://github.com/devemux86/cruiser/discussions/56) such as a cattle-grid or level-crossing (gate).
We currently have [a minor issue with BRouter](https://github.com/abrensch/brouter/issues/639) where it doesn't handle a motorcycle travelling toll-free on a road that permits it.

Today's learning-curve was both BaseCamp *and* Cruiser related...  For some reason, a road wasn't being included in a route and no matter where I placed a Shaping-Point to force the route, it just sent me back and around via a different road.  So, now being au fait with all things OSM and having made small fixes to `maxspeed` data, I decided to see why this road was so un-routable.

The road was fine. It permitted access and was a typical "Secondary" road comprising several segments in OSM.

Then I took a look at the junction - and I instantly saw the issue: It's a triangular layout with a small concrete island with a "keep left" sign. That ~10m stretch of road was tagged as `oneway` - but in the WRONG DIRECTION!  It had been like it for around 2yrs.

Here's the fixed version (with some additional arrows to show how the layout works) and which now permits travel from the right-hand-side:
![osm-oneway](https://github.com/devemux86/cruiser/assets/125696109/4811b89e-36c2-416c-bb6f-d926a64b99ce)

My edit is now reflected in BRouter and OSM (now compiled into a new BaseCamp map).  Every little helps :fist_raised:

**Comment by devemux86** [comment]
@MotoUKRider thanks for the report and the OpenStreetMap correction! :+1: 

There are many times when a wrong route calculation can be caused by incorrect OpenStreetMap tags.
If we know the right ones, it's nice to fix them so that the whole community can benefit from the result.

**Comment by Leeb64** [comment]
Unfortunately, there are quite a few map errors in the OSM maps.
I have often suspected the incorrect calculation in the APP. Most of the time it is a map error!
The only solution is to log in to OSM and make corrections.

**Reply by airald** [reply]
If there's an error not in the map, just in the directions or the permissions of the road, you can change it and it will be updated on Brouter on next day.

**Comment by MotoUKRider** [comment]
...and we're back - I've noticed a few issues locally with "odd" road definitions from OpenStreetMap which, in turn, cause BRouter/Cruiser to *not* offer a turn instruction.  Here are 3 examples:

This road runs left-right but changes from Primary to Secondary if you approach from the right. Therefore, if you make the right turn to "stay" on the road designation, you do *not* get a turn instruction.
![routing-fail-1](https://github.com/user-attachments/assets/1faa2d70-0561-4031-8a33-0f123e9b9484)

At this junction, you have to turn right to remain on the road, but there is no turn instruction:
![routing-fail-2](https://github.com/user-attachments/assets/48763565-b50f-460f-b9c3-c078b366e84c)

Saving the best for last: This is a cross-roads with the Secondary (left-right) road having priority. The Primary roads each have a "Stop" sign but the road has been draw as a continuous line:
![routing-fail-3](https://github.com/user-attachments/assets/a15ba42f-995a-4dcb-bb77-1cb3a31dedcc)

This is obviously an OpenStreetMap "thing" so I may try to correct them (I've only made simple edits like speed-limits/tolls) but highlights how sometimes the behaviour you expect isn't offered. It's probably more worrying on roads you are unfamiliar with!
